src/test/java/edu/kis/powp/jobs2d/TestJobs2dPatterns.java

```diff
@@ -43,6 +47,45 @@ private static void setupPresetTests(Application application) {
 		application.addTest("Figure Jane", e -> {
 			FiguresJane.figureScript(janeAdapter);
 		});
+
+		application.addTest("Primitives: Circle ", new ActionListener() {
+			private int radius = 5;
+			private int segments = 3;
+
+			@Override
+			public void actionPerformed(ActionEvent e) {
+				DriverCommand cmd = PrimitivesCommandFactory.createCircle(0, 0, radius, segments);
+				radius += 10;
+				segments += 1;
+
+				cmd.execute(driverManager.getCurrentDriver());
+			}
+		});
+
+		application.addTest("Primitives: Rectangle ", new ActionListener() {
+			private int pos = 5;
+
+			@Override
+			public void actionPerformed(ActionEvent e) {
+				DriverCommand cmd = PrimitivesCommandFactory.createRectangle(pos, pos, 50, 30);
+				pos += 10;
+				cmd.execute(driverManager.getCurrentDriver());
+			}
+		});
+
+		application.addTest("DriverCommand: Joe 1", e -> {
+			Job2dDriverToDriverCommandAdapter adapter = new Job2dDriverToDriverCommandAdapter();
+
+			FiguresJoe.figureScript1(adapter);
+			adapter.getCommand().execute(driverManager.getCurrentDriver());
+		});
+
+		application.addTest("DriverCommand: Joe 2", e -> {
+			Job2dDriverToDriverCommandAdapter adapter = new Job2dDriverToDriverCommandAdapter();
+
+			FiguresJoe.figureScript2(adapter);
+			adapter.getCommand().execute(driverManager.getCurrentDriver());
```

Comment:
Those tests should be in separate class. They make TestJobs2dPatterns harder to maintain and use.

It is good to keep a convention used in the original code (class instance instead of lambda)